src/process_utility.c

```diff
@@ -1697,9 +1697,58 @@ process_altertable_end_subcmd(Hypertable *ht, Node *parsetree, ObjectAddress *ob
 					 errmsg("hypertables do not support inheritance")));
 		case AT_SetStatistics:
 		case AT_SetLogged:
+		case AT_SetStorage:
+		case AT_ColumnDefault:
+		case AT_SetNotNull:
+		case AT_DropNotNull:
+		case AT_AddOf:
+		case AT_DropOf:
+#if PG10
+		case AT_AddIdentity:
+		case AT_SetIdentity:
+		case AT_DropIdentity:
+#endif
 			/* handled by default recursion */
 			break;
+		case AT_EnableRowSecurity:
+		case AT_DisableRowSecurity:
+		case AT_ForceRowSecurity:
+		case AT_NoForceRowSecurity:
+			/* should not recurse to chunks */
+			break;
+		case AT_ReAddConstraint:
+		case AT_ReAddIndex:
+		case AT_AddOidsRecurse:
+			/* internal commands correctly handled */
+			break;
+		case AT_AddColumn:
+		case AT_AddColumnRecurse:
+		case AT_DropColumn:
+		case AT_DropColumnRecurse:
+			/* handled in process_altertable_start_table */
+			break;
+		case AT_DropConstraint:
+		case AT_DropConstraintRecurse:
+			/* handled by process_ddl_sql_drop */
+			break;
+		case AT_ProcessedConstraint:	/* internal command never hit in our
+										 * test code, so don't know how to
+										 * handle */
+		case AT_ReAddComment:	/* internal command never hit in our test
+								 * code, so don't know how to handle */
+		case AT_AddColumnToView:	/* only used with views */
+		case AT_AlterColumnGenericOptions:	/* only used with foreign tables */
+		case AT_GenericOptions: /* only used with foreign tables */
+#if PG10
+		case AT_AttachPartition:	/* handled in
+									 * process_altertable_start_table but also
+									 * here as failsafe */
+		case AT_DetachPartition:
+#endif
 		default:
```

Comment:
I actually handle all possible enum values now. Wasn't sure whether to still keep "default" or not. Any thoughts?

Reply:
FWIW, if it is an `enum` type we are switching on, the compiler should complain if we remove the `default` case and there are types we do not handle. We can ensure this fails in, e.g., Travis. So, maybe we should remove the `default` case so that we will notice any new types being added?

So, to summarize, if we do list all types then it is probably better to _not_ have the `default` case.